src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+            }
+            else if (IsByteTrain(input, 4) == true)
+            {
+                return ColorStringTypes.Byte_longString;
+            }
+            else
+            {
+                return ColorStringTypes.None;
+            }
+        }
+
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte values, separated by a ",".
+        /// </summary>
+        /// <example>
+        /// 23,78,56=>true
+        /// 789,fp=>false
+        /// </example>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        private static bool IsByteTrain(string input, int num)
+        {
+            bool StrCheck = false;
+            string[] arrStr = input.Trim().Split(',');
+            if (num == arrStr.Length)
+            {
+                int i = 0;
+                foreach (var item in arrStr)
+                {
+                    if (Byte.TryParse(item, out byte result) == true)
+                    {
+                        i += 1;
+                    }
+                    else { continue; }
+                }
+                StrCheck = i == num ? true : false;
+            }
+            else
+            {
+                StrCheck = false;
+            }
+            return StrCheck;
+        }
+        
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte hex values,
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        /// <remarks><paramref name="input"/> should be left in the raw state(e.g.;#56e245, not 56e245).</remarks>
+        private static bool IsByteHexTrain(string input, int num)
+        {
+            bool StrCheck = false;
+            string subStr = input.Substring(1);
+            if (num == subStr.Length)
+            {
+                int c = 0;
+                for (int i = 1; i < num; i+=2)
+                {
+                    string item = subStr[i - 1].ToString() + subStr[1].ToString();
+                    if (Byte.TryParse(item, System.Globalization.NumberStyles.AllowHexSpecifier, CultureInfo.InvariantCulture, out byte result) == true)
+                    {
+                        c+=2;
+                    }
+                    else { continue; }
+                }
+ 
+                StrCheck = (c == num )? true : false;
+            }
+            else
+            {
+                StrCheck = false;
+            }
+            return StrCheck;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="byte"/> values from the <paramref name="input"/> if the number of items are ==<paramref name="num"/>.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        private static byte[] ByteStringValues(string input,int num)
```

Comment:
I don't like the separation of `IsByteTrain` and `ByteStringValues`, the same goes for ther hex counterparts. Could we make two functions here: one for "bytes" and one for "hex"? Let it do validation and extraction in one pass.

Reply:
I've done that in the new branch

Reply:
That branch is named `Feature-Hex-codes`, right?

Reply:
Yes